### PR TITLE
add gap to theme for button

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
Adds the ability to change the gap through the theme 
#### Where should the reviewer start?
button.js
#### What testing has been done on this PR?
tested it using a value in the theme 
#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
Design System needs a way to change `gap` to `6px` for buttons with Icons
#### Do the grommet docs need to be updated?
yes
#### Should this PR be mentioned in the release notes?
sure
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/Button/Button.js [hunk 1]
```diff
@@ -276,7 +276,7 @@ const Button = forwardRef(
           direction="row"
           align="center"
           justify={justify || (align === 'center' ? 'center' : 'between')}
-          gap={gap}
+          gap={theme.button.gap || gap}
```
Comment (taysea) [inline src/js/components/Button/Button.js:279]:
```suggestion
          gap={gap || theme.button.gap}
```
should it be like this? theme would be fallback if gap prop is undefined

File: src/js/themes/base.js [hunk 2]
```diff
@@ -418,6 +418,7 @@ export const generate = (baseSpacing = 24, scale = 6) => {
           },
         },
       },
+      // gap: undefined,
```
Comment (taysea) [inline src/js/themes/base.js:421]: Should the default `small` from Button.js get moved here to the theme?

Comment (britt6612) [reply src/js/themes/base.js:421]: I was debating on this as well and removing from the prop I just wasn't sure

**Review by taysea — APPROVED** [review]
LGTM!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**